api/v1beta2/foundationdbcluster_types.go

```diff
@@ -2275,3 +2275,18 @@ func (cluster *FoundationDBCluster) Validate() error {
 
 	return fmt.Errorf(strings.Join(validations, ", "))
 }
+
+// ContainsPod checks if the given Pod is part of the cluster or not.
+func (cluster *FoundationDBCluster) ContainsPod(pod corev1.Pod) bool {
```

Comment:
Is that something we want to expose from the cluster struct of should this be rather part of the internal package? What do you think?